web/src/main/webapp/ui/menu.ts

```diff
@@ -282,6 +431,11 @@ export class SubMenu extends BaseMenu<MenuItem> implements IHtmlElement {
         else
             cell.onclick = (e: MouseEvent) => { e.stopPropagation(); this.hide(); };
     }
+
+    public setSubMenuAction(parentIndex: number, subMenuIndex: number,
```

Comment:
rather than have this method be abstract, it's better to make the base method throw.